service/src/main/kotlin/fi/espoo/evaka/decision/Decision.kt

```diff
@@ -30,23 +31,51 @@ data class Decision(
     val requestedStartDate: LocalDate?,
     val resolved: LocalDate?
 ) {
-    fun validRequestedStartDatePeriod() = FiniteDateRange(
+    fun validRequestedStartDatePeriod(featureConfig: FeatureConfig) = FiniteDateRange(
         startDate,
         minOf(
             endDate,
             startDate.plusDays(
                 when (this.type) {
-                    DecisionType.CLUB -> 14
-                    DecisionType.DAYCARE, DecisionType.DAYCARE_PART_TIME -> 14
+                    DecisionType.CLUB -> featureConfig.requestedStartUpperLimit
+                    DecisionType.DAYCARE, DecisionType.DAYCARE_PART_TIME -> featureConfig.requestedStartUpperLimit
                     DecisionType.PRESCHOOL -> 0
-                    DecisionType.PRESCHOOL_DAYCARE -> 14
+                    DecisionType.PRESCHOOL_DAYCARE -> featureConfig.requestedStartUpperLimit
                     DecisionType.PREPARATORY_EDUCATION -> 0
-                }
+                }.toLong()
             )
         )
     )
+
+    fun toDecisionWithValidStartDatePeriod(featureConfig: FeatureConfig) = DecisionWithValidStartDatePeriod(
+        id = id,
+        type = type,
+        startDate = startDate,
+        endDate = endDate,
+        unit = unit,
+        applicationId = applicationId,
+        childId = childId,
+        childName = childName,
+        sentDate = sentDate,
+        status = status,
+        validRequestedStartDatePeriod = validRequestedStartDatePeriod(featureConfig)
+    )
 }
 
+data class DecisionWithValidStartDatePeriod(
+    val id: DecisionId,
+    val type: DecisionType,
+    val startDate: LocalDate,
+    val endDate: LocalDate,
+    val unit: DecisionUnit,
+    val applicationId: ApplicationId,
+    val childId: ChildId,
+    val childName: String,
+    val sentDate: LocalDate,
+    val status: DecisionStatus,
+    val validRequestedStartDatePeriod: FiniteDateRange
+)
+
```

Comment:
We have talked about minimizing data classes that are almost identical. How about having something like `{ decision: Decision, validRequestedStartDatePeriod: FiniteDaterange }` to avoid copy-pasting all those fields?

An even better option would be to just add `validRequestedStartDatePeriod` to `Decision`, but that would need bigger changes to propagate `featureConfig` to every place where `Decision` instances are created.